src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -599,6 +599,34 @@ protected function whereJsonContains(Builder $query, $where)
         );
     }
 
+    /**
+     * Compile an "is distinct from" clause.
+     *
+     * @param  \Illuminate\Database\Query\Builder  $query
+     * @param  array  $where
+     * @return string
+     */
+    protected function whereDistinctFrom(Builder $query, $where)
+    {
+        $column = $this->wrap($where['column']);
+
+        return '('.$column.' != '.$this->parameter($where['value']).' or '.$column.' is null)';
```

Comment:
I  think this should probably use the [null safe equals operator](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_equal-to).

> The [<=>](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_equal-to) operator is equivalent to the standard SQL IS NOT DISTINCT FROM operator

So effectively, using `NOT col1 <=> col2` is equivalent to using `IS DISTINCT FROM` in other SQL dialects.

Reply:
Interesting 👀

Reply:
I just realized that this is the default grammar and not MySQL specific.  The MySQL docs make me think `IS DISTINCT FROM` may be the SQL standard and MySQL is a little weird with different syntax.  Maybe we could move the PostgreSQL version here and update the `MySqlGrammar` instead of `PostgresGrammar` since the postgres  mechanism is more in line with the SQL standard.  Or alternatively, we could update `MySqlGrammar` and leave this and the `PostgresGrammar`  in place as is?

Reply:
Seems like `is distinct from` only works on PostgreSQL, `not <=>` only on MySQL and MariaDB and the rest need to use `!= or is null` 🤷